app/src/handlers/census/events/DeathEvent.ts

```diff
@@ -79,9 +75,9 @@ export default class DeathEvent {
 
         // No check needed, ZoneUtils will take care of this
         this.zone = ZoneUtils.parse(Parser.parseNumericalArgument(event.zone_id));
-        this.timestamp = Parser.parseNumericalArgument(event.timestamp);
+        this.timestamp = event.timestamp;
 
-        if (isNaN(this.timestamp)) {
+        if (this.timestamp === undefined || this.timestamp === null) {
```

Comment:
`event.timestamp` is guaranteed to have a Date object, else the client will emit an error through warn.

Reply:
@Maelstromeous we currently validate everything we get from ps2census (at least for null and undefined). Should we drop this?

Reply:
Yes, it's unnessessary since this DeathEvent will always be hydrated. If not, errors will occur and we'll detect them.

Reply:
Then we can just drop all the event classes?

Reply:
No, we still require them for calculating logic and further injection of additional data, e.g. the ActiveAlert property.